dds/DCPS/XTypes/DynamicDataImpl.cpp

```diff
@@ -7,3263 +7,6479 @@
-    if (member_type->get_descriptor(member_td) != DDS::RETCODE_OK) {
-      if (DCPS::DCPS_debug_level >= 1) {
-        ACE_DEBUG((LM_DEBUG, ACE_TEXT("(%P|%t) DynamicDataImpl::get_from_struct_common_checks -")
-                   ACE_TEXT(" Could not get type descriptor for member %d\n"),
-                   id));
-      }
-      return false;
-    }
-    const TypeKind elem_kind = get_base_type(member_td->element_type())->get_kind();
-    if (elem_kind != kind) {
-      if (DCPS::DCPS_debug_level >= 1) {
-        ACE_DEBUG((LM_DEBUG, ACE_TEXT("(%P|%t) DynamicDataImpl::get_from_struct_common_checks -")
-                   ACE_TEXT(" Member with ID %d is a sequence of %C, not %C\n"),
-                   id, typekind_to_string(elem_kind), typekind_to_string(kind)));
-      }
-      return false;
-    }
-  }
+DynamicDataImpl::SequenceValue::SequenceValue(const DDS::Float64Seq& float64_seq)
+  : elem_kind_(TK_FLOAT64)
+{
+  active_ = new(float64_seq_) DDS::Float64Seq(float64_seq);
+}
 
-  return true;
+DynamicDataImpl::SequenceValue::SequenceValue(const DDS::Float128Seq& float128_seq)
+  : elem_kind_(TK_FLOAT128)
+{
+  active_ = new(float128_seq_) DDS::Float128Seq(float128_seq);
 }
 
-bool DynamicDataImpl::skip_struct_member_at_index(ACE_CDR::ULong index, ACE_CDR::ULong& num_skipped)
+DynamicDataImpl::SequenceValue::SequenceValue(const DDS::CharSeq& char8_seq)
+  : elem_kind_(TK_CHAR8)
 {
-  const DDS::DynamicType_var base_type = get_base_type(type_);
+  active_ = new(char8_seq_) DDS::CharSeq(char8_seq);
+}
 
-  DDS::DynamicTypeMember_var member;
-  if (base_type->get_member_by_index(member, index) != DDS::RETCODE_OK) {
-    if (DCPS::DCPS_debug_level >= 1) {
-      ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) DynamicDataImpl::skip_struct_member_at_index -")
-                 ACE_TEXT(" Failed to get DynamicTypeMember for member index %d\n"), index));
-    }
-    return false;
-  }
+DynamicDataImpl::SequenceValue::SequenceValue(const DDS::ByteSeq& byte_seq)
+  : elem_kind_(TK_BYTE)
+{
+  active_ = new(byte_seq_) DDS::ByteSeq(byte_seq);
+}
 
-  DDS::MemberDescriptor_var md;
-  if (member->get_descriptor(md) != DDS::RETCODE_OK) {
-    return false;
-  }
-  if (strm_.encoding().kind() == DCPS::Encoding::KIND_XCDR2 && md->is_optional()) {
-    ACE_CDR::Boolean present;
-    if (!(strm_ >> ACE_InputCDR::to_boolean(present))) {
-      return false;
-    }
-    // Optional member that is omitted is not counted as a skipped member.
-    if (!present) {
-      num_skipped = 0;
-      return true;
-    }
-  }
+DynamicDataImpl::SequenceValue::SequenceValue(const DDS::BooleanSeq& boolean_seq)
+  : elem_kind_(TK_BOOLEAN)
+{
+  active_ = new(boolean_seq_) DDS::BooleanSeq(boolean_seq);
+}
 
-  num_skipped = 1;
-  const DDS::DynamicType_ptr member_type = md->type();
-  return member_type && skip_member(member_type);
+// TODO: Is this sufficient for string and wstring? or does it need to duplicate each
+// string/wstring in the sequence?
```

Comment:
Sequence objects own their contents, so StringSeq acts just like `vector<string>`.  The TODO can probably be removed.